client_apis/python/src/tests/integration/alerts_test.py

```diff
@@ -0,0 +1,212 @@
+    @classmethod
+    def setUpClass(cls):
+        cls.watchlist_name = "alerts_watchlist_%s" % datetime.now().strftime('%m%d%Y%H%M%S')
+
+        watchlist = cb.watchlist_add(
+            type="events",
+            name=cls.watchlist_name,
+            search_query="b.urlver=1&q=process_name%3Achrome.exe&sort=&rows=10&start=0",
+            readonly=False
+        )
+
+        cls.watchlist_id = watchlist["id"]
+
+        cb.watchlist_action_add(
+            watchlist_id=cls.watchlist_id,
+            action_type_id=3,
+            email_recipient_user_ids=[1]
+        )
+
+        ShhHelper.execute_command(server, user, password, '/usr/bin/python -m cb.maintenance.job_runner --master -s watchlist_search')
+        time.sleep(20)
+        ShhHelper.execute_command(server, user, password, '/usr/share/cb/cbsolr --soft-commit')
+        time.sleep(5)
+
+    def test_all_positive_cases(self):
+
+        # Get All Alerts
+        alerts = cb.alert_search('')
+        self.assertIsNotNone(alerts)
+        self.assertIsNotNone(alerts['results'])
+        self.assertNotEqual(len(alerts['results']), 0)
+        # print json.dumps(alerts, indent=4, sort_keys=True)
+
+        # Search For A Specific Alert
+        alerts = cb.alert_search(CbApiAlertTestCase.watchlist_name)
+        self.assertIsNotNone(alerts)
+        self.assertIsNotNone(alerts['results'])
+        self.assertNotEqual(len(alerts['results']), 0)
+        alert_to_update = alerts['results'][0]
+        #print json.dumps(alert_to_update, indent=4, sort_keys=True)
+
+        # "Search For A Non Existing Alert
+        alerts = cb.alert_search('watch_list_non')
+        self.assertIsNotNone(alerts)
+        self.assertEqual(len(alerts['results']), 0)
+
+        # Update Alert
+        alert_to_update['status'] = 'resolved'
+        result = cb.alert_update(alert_to_update)
+        self.assertIsNotNone(result)
+        self.assertEqual(result['result'], 'success')
+        #print json.dumps(result, indent=4, sort_keys=True)
+
+        time.sleep(20)
+        alerts = cb.alert_search(CbApiAlertTestCase.watchlist_name)
+        self.assertIsNotNone(alerts)
+        self.assertIsNotNone(alerts['results'])
+        self.assertNotEqual(len(alerts['results']), 0)
+
+        for alert in alerts['results']:
+            if alert['unique_id'] == alert_to_update['unique_id']:
+                self.assertEqual(alert['status'], 'Resolved')
+
+        return
+
+    def test_negative_cases(self):
+        # /api/v1/alert/<alert_id>
+
+        # Update Alert with wrong id in the path
+        alerts = cb.alert_search(CbApiAlertTestCase.watchlist_name)
+        self.assertIsNotNone(alerts)
+        self.assertIsNotNone(alerts['results'])
+        self.assertNotEqual(len(alerts['results']), 0)
+        alert = alerts['results'][0]
+
+        token_header = {'X-Auth-Token': token}
+
+        r = requests.post("%s/api/v1/alert/%s" % (url, alert['unique_id'] + '123'), headers=token_header,
```

Comment:
It would be an exceptional case where we'd want the tests to invoke HTTP requests directly rather than go through `cbapi.py`.  Change these calls to go through methods on `CbApi`, and if there aren't currently any matching methods, please create them.